proto_text.c

```diff
@@ -1460,10 +1476,20 @@ static void process_mset_command(conn *c, token_t *tokens, const size_t ntokens)
             comm = NREAD_ADD;
             break;
         case 'A': // Append.
-            comm = NREAD_APPEND;
+            if (of.vivify) {
+                comm = NREAD_APPENDVIV;
+                exptime = of.autoviv_exptime;
+            } else {
+                comm = NREAD_APPEND;
+            }
             break;
         case 'P': // Prepend.
-            comm = NREAD_PREPEND;
+            if (of.vivify) {
+                comm = NREAD_PREPENDVIV;
+                exptime = of.autoviv_exptime;
```

Comment:
what takes the precedence? is it `T` or `N` taking precedence?

Reply:
T is ignored when append/prepend are used. That's unfortunately gross but the semantics don't work right otherwise.

Reply:
well - to rephrase the question - I just want to know about: 
1. if item exists, memcached will prefer `T`
2. it item gets vivified, memcached will prefer `N`

don't know all the internals of memcached wrt other commands and want to ensure that this new change uses the same approach.

Reply:
oh - so continuing to use `T` wouldn't make any difference to the item, and have the first exptime used?

Reply:
That's how it works with the rest of the commands. N initializes and T updates (or sets if used directly). I haven't done a code review in case I goofed it somehow but I think it's right?

Reply:
okay - I don't know how for sure and just wanted to ensure it's the same approach

Reply:
I did a small test from your build and not sure if this is expected:
```log
hemal@qa-mydev--0ad632f415f692eb1.northwest.stripe.io:/pay/src$ telnet localhost 11211
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
ms append-test 5 MA N100 T100
12345
HD
ms append-test 5 MA N100 T100
67890
HD
mg append-test v
VA 10
1234567890
mg append-test v t
VA 10 t56
1234567890
ms append-test 5 MA N100 T100
2
22
HD
mg append-test v t
VA 15 t38
12345678902
22
```

It doesn't seem to take the `T100` value in effect

Reply:
It's just specifically weird for append. T is completely ignored for append because it normally can't create a new item. If you add the N flag, it _can_ create a new item, but uses the TTL from the token N requires. For mg and ma cases N and T both initialize or update. In a perfect world I would have set use the same semantics, but I didn't and we have to live with it :)

If set used N and T it would be trickier to use anyway; you'd often have to set both flags, or we'd need an exception for set to say T works like N if missing and you're in set mode and etc. Then it's just all complicated.

We might get an extra win if append with T can update the TTL but that's out of scope for this change.

Reply:
okay - that's what you meant by previous comment - its now more clearer. I think in this case we always would have to end up using the `N` flag as the final value instead of keep on extending the TTL

Reply:
or we can just do `mg <key> T<extendTTL> time I guess

Reply:
You've confused me enough that I'm leaving a note here for me to double check the T semantics for append and such. :)

Reply:
Double checked: append code explicitly reuses the expiration time of the original item when appending. Making T update TTL on append sounds nice but can be its own change; for now it does nothing.